onnxruntime/core/session/inference_session.cc

```diff
@@ -2386,34 +2386,31 @@ common::Status InferenceSession::RunAsync(const RunOptions* run_options,
                                           RunAsyncCallbackFn callback,
                                           void* user_data) {
   size_t num_fetches = fetch_names.size();
-  if (!thread_pool_.get() || concurrency::ThreadPool::DegreeOfParallelism(thread_pool_.get()) < 2) {
+  auto* tp = intra_op_thread_pool_from_env_ ? intra_op_thread_pool_from_env_ : thread_pool_.get();
```

Comment:
It looks like we've [3 sets](https://github.com/microsoft/onnxruntime/blob/e1ca8ee6d4a188f01efd4083d6f4273a3ef901aa/onnxruntime/core/session/inference_session.h#L709-L719) of threadpools. The threadpool that gets [passed](https://github.com/microsoft/onnxruntime/blob/a4902ee65b2899a247ea85e07c69b253ef6fa11f/onnxruntime/core/session/inference_session.cc#L1460-L1469) to session state comes from the function [GetIntraOpThreadPoolToUse](https://github.com/microsoft/onnxruntime/blob/e1ca8ee6d4a188f01efd4083d6f4273a3ef901aa/onnxruntime/core/session/inference_session.h#L536-L549) which seems to completely disregard the global threadpool. So, this might be a bug. Secondly, we should rely on the GetIntraOpThreadPoolToUse function here to return the correct threadpool to use.

Reply:
Done.